bindings/python/gnucash_core.py

```diff
@@ -451,6 +455,144 @@ def __args_to_instance(args):
         else:
             raise TypeError('Required single int/float/str or two ints: ' + str(args))
 
+    # from https://docs.python.org/3/library/numbers.html#numbers.Integral
+    # and https://github.com/python/cpython/blob/3.7/Lib/fractions.py
+
+    def _operator_fallbacks(monomorphic_operator, fallback_operator):
+        def forward(a, b):
+            if isinstance(b, GncNumeric):
+                return monomorphic_operator(a, b)
+            if isinstance(b, int):
+                return monomorphic_operator(a, GncNumeric(b))
+            elif isinstance(b, float):
+                return monomorphic_operator(a, GncNumeric(b))
+            else:
+                return NotImplemented
+        forward.__name__ = '__' + fallback_operator.__name__ + '__'
+        forward.__doc__ = monomorphic_operator.__doc__
+
+        def reverse(b, a):
+            if isinstance(a, GncNumeric):
+                return monomorphic_operator(a, b)
+            elif isinstance(a, int):
+                return monomorphic_operator(GncNumeric(a), b)
+            elif isinstance(a, float):
+                return fallback_operator(float(a), float(b))
```

Comment:
The way you suggest an arithmetic operation with an GncNumeric will result in another GncNumeric. It seemed right to me to allow a float as a result depending on the position.

Your suggestion: Float X GncNumeric -> GncNumeric; GncNumeric X Float -> GncNumeric
My suggestion: Float X GncNumeric -> Float; GncNumeric X Float -> GncNumeric

where X is in *,/,+,.

I've checked the relation of Decimals and Fractions to Floats.
Fractions in arithmetic relations to float always turn to Float.
Decimals do not support arithmetics with floats at all.

I suppose there is no "standard" behavior. It might even be easier to know that arithmetics with GncNumeric will be "greedy" and always result in GncNumeric. I'll go your way.

Reply:
Yeah, Decimals are doing it right, floats are evil. ;-) Seriously, floats have no place in financial applications because they can't represent decimal fractions exactly. Try it out in the interpreter:
```
>>> num = 1 / 10
>>> print ('%.32f' % num)
0.10000000000000000555111512312578
```

Multiply a couple of those and you have an imbalance.

Reply:
@jralls I only realize this when working on this topic. I think it would be important to have this issue documented throughly somewhere and the approaches in the GnuCash framework to address it. Be it in the C-Source docs, Python docstrings, a general doxygen page or the wiki or connect the existing bits of information to make it more visible.

Reply:
Passing a string to GncNumeric provides the expected result:

```
>>> print(GncNumeric("2.2"))                                                                                                                 
22/10
```

Reply:
Can GncNumeric be initialised with a float in that way that the insignificances get rounded away? Can the significant numbers be provided in initialisation of GncNumeric?

Something like `GncNumeric(2.2, denom=100)`?

Reply:
```
>>> GNC_HOW_DENOM_SIGFIG = 0x50
>>> n=2
>>> for n in range(0,20): 
    ...:     print(GncNumeric(2.2, gnucash.GNC_DENOM_AUTO, GNC_HOW_DENOM_SIGFIG | n * 0x100)) 
    ...:                                                                                                                                               
2/1
2/1
22/10
220/100
2200/1000
22000/10000
220000/100000
2200000/1000000
22000000/10000000
220000000/100000000
2200000000/1000000000
22000000000/10000000000
220000000000/100000000000
2200000000000/1000000000000
22000000000000/10000000000000
220000000000000/100000000000000
22000000000000002/10000000000000000
220000000000000025/100000000000000000
2200000000000000256/1000000000000000000
2200000000000000256/1000000000000000000
```
GNC_HOW_DENOM_SIGFIG is missing in gnucash. I'll add it.

Reply:
Another way to achieve this:
```
>>> print(gnucash.GncNumeric(2.2).convert(1000, gnucash.GNC_HOW_RND_ROUND))
2200/1000
```

Reply:
If only it was that easy. That approach fails when either the ordinal or the number of decimals gets large. [This lecture](https://www.youtube.com/watch?v=4P_kbF0EbZM) gets into the gory details.

> I think it would be important to have this issue documented throughly somewhere and the approaches in the GnuCash framework to address it. 

For the Doxygen docs the most logical place would be [Query-object-framework>Nuimeric](https://code.gnucash.org/docs/MAINT/group__Numeric.html) but I'm not sure that that's sufficiently discoverable. Something in the Wiki makes sense: what were the first two pages you looked at when you thought of it?

Reply:
I searched for GncNumeric (leads to https://wiki.gnucash.org/wiki/C%2B%2B) in the wiki and gnc_numeric (leads to https://wiki.gnucash.org/wiki/Using_the_API).

Just now I'm creating a commented python example file for GncNumeric. Btw I am curious about Jupyter notebooks and wonder if they could be useful but that just as a sidenote.

The page https://code.gnucash.org/docs/MAINT/group__Numeric.html is a good point. Could be extended a bit. A wiki page makes sense pointing at this doxygen page so that it is easier to find. Should a wiki page be about the general issues of handling numbers, the GnuCash-engine aspects or the Python Wrapper perspective?

Reply:
Keep in mind that the C++ GncNumeric is the implementation underneath gnc_numeric, while the Python GncNumeric is a wrapper around gnc_numeric.

[Using the API](https://wiki.gnucash.org/wiki/Using_the_API) seems like a good place for cautions about both floating-point numbers and GNC_DENOM_SIGFIG, except that it doesn't go into any kind of detail and while it does point to the Doxygen docs it correctly  doesn't have much nice to say about them.  The Python bindings have [their own page](https://wiki.gnucash.org/wiki/Python_Bindings)that could also use some sprucing up too.